SpatialGDK/Source/SpatialGDK/Public/Utils/SpatialInterestFactory.h

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Interop/SpatialClassInfoManager.h"
+#include "Schema/Interest.h"
+
+#include <WorkerSDK/improbable/c_worker.h>
+
+/**
+ * The SpatialInterestFactory is responsible for creating spatial Interest component state and updates for a GDK game.
+ *
+ * It has two dependencies:
+ *   - the class info manager for finding level components and for creating user defined queries from ActorInterestComponents
+ *   - the package map, for finding unreal object references as part of creating AlwaysInterested constraints
+ *     (TODO) remove this dependency when/if we drop support for the AlwaysInterested constraint
+ *
+ * The interest factory is initialized within and has its lifecycle tied to the spatial net driver.
+ *
+ * There are two public types of functionality for this class.
+ *
+ * The first is actor interest. The factory takes information about an actor (the object, info and corresponding entity ID)
+ * and produces an interest data/update for that entity. This interest contains anything specific to that actor, such as self constraints
+ * for servers and clients, and if the actor is a player controller, the client worker's interest is also built for that actor.
+ *
+ * The other is server worker interest. Given a load balancing strategy, the factory will take the strategy's defined query constraint
+ * and produce an interest component to exist on the server's worker entity. This interest component contains the primary interest query made
+ * by that server worker.
+ */
+
+class UAbstractLBStrategy;
+class USpatialClassInfoManager;
+class USpatialPackageMapClient;
+
+DECLARE_LOG_CATEGORY_EXTERN(LogInterestFactory, Log, All);
+
+namespace SpatialGDK
+{
+class SPATIALGDK_API SpatialInterestFactory
```

Comment:
Lets just call it `InterestFactory`. We generally only add Spatial to things that have a native version.